--- start-tor-browser	2013-07-05 16:03:51.274599501 -0700
+++ start-tor-browser	2013-07-05 16:11:04.451868767 -0700
@@ -209,6 +209,8 @@
 export LD_LIBRARY_PATH
 
 if [ "$debug" -eq 1 ]; then
+        cd "${HOME}/App/fte_relay"
+        ./bin/fte_relay --mode client &
 	printf "\nStarting Vidalia now\n"
 	cd "${HOME}"
 	printf "\nLaunching Vidalia from: `pwd`\n"
@@ -217,15 +219,21 @@
 	./App/vidalia --loglevel debug --logfile vidalia-debug-log \
 	--datadir Data/Vidalia/ -style Cleanlooks
 	printf "\nVidalia exited with the following return code: $?\n"
+        cd "${HOME}/App/fte_relay"
+        ./bin/fte_relay --mode client --stop
 	exit
 fi
 
 # not in debug mode, run proceed normally
 printf "\nLaunching Tor Browser Bundle for Linux in ${HOME}\n"
+cd "${HOME}/App/fte_relay"
+./bin/fte_relay --mode client &
 cd "${HOME}"
 # XXX Someday we should pass whatever command-line arguments we got
 # (probably filenames or URLs) to Firefox.
 ./App/vidalia --datadir Data/Vidalia/ -style Cleanlooks
+cd "${HOME}/App/fte_relay"
+./bin/fte_relay --mode client --stop
 exitcode="$?"
 if [ "$exitcode" -ne 0 ]; then
 	complain "Vidalia exited abnormally.  Exit code: $exitcode"
